src/Unity.Mathematics.Editor/PrimitiveVectorDrawer.cs

```diff
@@ -89,9 +89,141 @@ public override void OnGUI(Rect position, SerializedProperty property, GUIConten
             if (attribute is DoNotNormalizeAttribute && string.IsNullOrEmpty(label.tooltip))
                 label.tooltip = Content.doNotNormalizeTooltip;
 
-            EditorGUI.BeginProperty(position, label, property);
-            EditorGUI.MultiPropertyField(position, subLabels, property.FindPropertyRelative(startIter), label);
+            label = EditorGUI.BeginProperty(position, label, property);
+            EditorGUICopy.MultiPropertyField(position, subLabels, property.FindPropertyRelative(startIter), label);
             EditorGUI.EndProperty();
         }
     }
+
+    internal class EditorGUICopy
+    {
+        internal const float kSingleLineHeight = 18f;
+        internal const float kSpacingSubLabel = 4;
+        private const float kIndentPerLevel = 15;
+        internal const float kPrefixPaddingRight = 2;
+        internal static int indentLevel = 0;
+        private static readonly int s_FoldoutHash = "Foldout".GetHashCode();
+
+        // internal static readonly SVC<float> kVerticalSpacingMultiField = new SVC<float>("--theme-multifield-vertical-spacing", 0.0f);
+        // kVerticalSpacingMultiField should actually look like the above line ^^^ but we don't have access to SVC<T>,
+        // so instead we just set this value to what is observed in the debugger with the Unity dark theme.
+        internal const float kVerticalSpacingMultiField = 2;
```

Comment:
It's a bit dicey since this decouples it from the stylesheets. So when design decides to change this throughout the editor, or when they need to create alternative stylesheets for e.g., vision-impaired users, it won't get picked up here. (To wit, it's a bit of a red flag to me that our default value for this is 0, but it sounds like it is currently actually 2 in the stylesheet.)

This again comes back to the issues with circumventing a fix at the root of the problem. (Because e.g., we could add an API like `multiFieldVerticalSpacing` right next to [`EditorGUIUtility.standardVerticalSpacing`](https://github.com/Unity-Technologies/UnityCsReference/blob/61f92bd79ae862c4465d35270f9d1d57befd1761/Editor/Mono/EditorGUIUtility.cs#L565) but then you'd have to do the whole backport/ifdef song-and-dance anyway, in which case we may as well just go back and fix `MultiPropertyField()` instead.

Reply:
@pcosgrave @amechtley Maybe we should do this instead:

1. Make the internals visible to Unity.Mathematics.Editor, but only in the newest versions of Unity. Don't backport that change.
2. In Mathematics, I can ifdef the code. For all the older versions of the editor, we use this non-ideal workaround with the duplicated code. But the newer versions that have the internals exposed, I will call the internal method directly.

Something like this:

```csharp
#if UNITY_2022_1_OR_NEWER
  EditorGUI.MultiPropertyFieldInternal(...);
#else
  EditorGUICopy.MultiPropertyField(...);
#endif
```

This way, we can fix the inspector drawing issue for the older editors and just accept that any changes to the stylesheets won't be seen by this code. But for the newer versions of the editor, since I'll be calling directly into the GUI code, the workaround isn't used and we won't have to worry about the duplicated code going out of sync with anything.

Reply:
That contravenes [this standard](https://github.cds.internal.unity3d.com/unity/standards/blob/8c81f4477fce9dfbed48276bb7f9254ff240893b/definitions/US-0010/US-0010.md), so RM would likely not allow it without question. It may require some coordination with e.g., @pcosgrave and @mathieum-unity , but they may permit it given the effort/benefit ratio, if there's a plan to track it as a bug and fix it asap once it is possible to resolve this with the replacement control in UITK. Either way, you'd need to have a conversation with them, presumably @carlosmauriciorincon

Reply:
The plan now is to add overloads of `MultiPropertyField` which has the `PropertyVisibility` enum as a parameter so I can call into the function directly. For newer editor versions, I'll call those overloads and for the old ones, I'll continue to use this duplicated code workaround. This PR is updated to handle the case when these new method overloads are available.